vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,166 @@
+---
+title: vcpkg_make_configure
+description: Use vcpkg_make_configure to configure a Make-based project.
+ms.date: 01/30/2024
+---
+# vcpkg_make_configure
+
+Configure a Make-based project.
+
+## Usage
+        
+```cmake
+vcpkg_make_configure(
+    [AUTOCONFIG]
+    [COPY_SOURCE]
+    [DISABLE_MSVC_WRAPPERS]
+    [NO_CPPFLAGS]
+    [ADD_BIN_TO_PATH]
+    [NO_DEFAULT_OPTIONS]
+    [NO_MSVC_FLAG_ESCAPING]
+    [USE_RESPONSE_FILES]
+    [SOURCE_PATH <source-path>]
+    [OPTIONS <configure-setting>...]
+    [OPTIONS_RELEASE <configure-setting>...]
+    [OPTIONS_DEBUG <configure-setting>...]
+    [PRE_CONFIGURE_CMAKE_COMMANDS <cmake-command>...]
+    [LANGUAGES <language>...]
+)
+```
+
+To use this function, you must depend on the helper port `vcpkg-make`:
+
+```json
+"dependencies": [
+  {
+    "name": "vcpkg-make",
+    "host": true
+  }
+]
+```
+
+## Parameters
+
+### AUTOCONFIG
+
+Runs autoreconf when set.
+
+### COPY_SOURCE
+
+If specified, the source directory will be copied to the build directory before the configuration process begins. Useful for projects that modify the source tree during configuration or build.
+
+### DISABLE_MSVC_WRAPPERS
+
+Disables the use of wrapper scripts for MSVC tools (cl.exe, windres).
+These wrappers translate and reorder command-line flags to accommodate the expected syntax and operational requirements of these tools. When this option is enabled, the `cl.exe` and `windres` tools will directly interpret command-line flags without any modifications.
+
+### NO_CPPFLAGS
+
+Disables the extraction of C preprocessor flags (`-D` and `-isystem`) from `CFLAGS` into `CPPFLAGS`.
+
+By default (when `NO_CPPFLAGS` is not set or is false), vcpkg automatically identifies and extracts these flags from the `CFLAGS` and `CXXFLAGS` variables, which typically contain general compiler options. It then consolidates these specific flags into a separate `CPPFLAGS` variable, dedicated to preprocessor and include path settings.
+
+However, if you explicitly set `NO_CPPFLAGS` to true, you override this default behavior. Vcpkg will then leave the `-D` and `-isystem` flags untouched in their original variables (`CFLAGS` and `CXXFLAGS`). 
+
+If set, the `CPPFLAGS` environment variable, typically used for specifying C preprocessor flags, will not be automatically populated based on vcpkg's settings. This can be useful for projects that do not use standard environment variables or have specific preprocessor requirements.
+
+### ADD_BIN_TO_PATH
+
+Adds the configure dependent `(debug/)bin` directory to the system path. This is useful if configure builds and runs executables with vcpkg integrated dependencies.
```

Comment:
What does 'configure dependent' mean?

Hmmm... what does this do when crosscompiling?